php/proxy.php

```diff
@@ -3,150 +3,192 @@
  * Acrolinx Proxy
  * Description - Creates proxy for Acrolinx.
  * Author - Acrolinx
- * Version - 1.0
+ * Version - 1.1
  */
- 
+
 class Proxy {
+
     private $settings;
     private $targetPath;
     private $username;
     private $password;
-    private $domainURL;	
-	
+    private $domainURL;
+    private $host;
+    private $url;
+
     private function disable_gzip() {
         @ini_set('zlib.output_compression', 'Off');
-        @ini_set('output_handler', '');	
+        @ini_set('output_handler', '');
     }
-    
-    
+
     private function getSettings() {
         $acrolinxSettings = array();
-
         $acrolinxSettings['isSSO'] = 1; //TODO: Set if SSO should be used, otherwise it will be just a reverse proxy.
         $acrolinxSettings['url'] = ""; //TODO: Set Acrolinx server url here.
         $acrolinxSettings['genericPassword'] = "secret"; //TODO: Set secret token here.
 
         if($acrolinxSettings['isSSO'] == 1) {
-
             $acrolinxSettings['password'] = $acrolinxSettings['genericPassword'];
             $acrolinxSettings['username'] = get_current_user(); //TODO: Implement get_current_user, which retrieves the username from the current session.
         }
         return $acrolinxSettings;
     }
-    
+
     private function getPosition($requestURI, $part) {
         return strpos($requestURI, $part);
     }
-    
+
     private function getActualURI($requestURI, $pos, $part) {
         return substr_replace($requestURI, '', 0, ($pos+strlen($part)));
     }
-    
+
+    private function emulate_getallheaders() {
+        foreach ($_SERVER as $name => $value) {
+            if (substr($name, 0, 5) == 'HTTP_') {
+                $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
+                $headers[$name] = $value;
+            } else if ($name == "CONTENT_TYPE") {
```

Comment:
When does this happen? Why not to lower in this case?

Reply:
This is needed while making a POST request like(http://localhost/acrolinx/proxy.php/iq/services/v1/rest/login). In general for uploading content. If this is not done error 415(Unsupported media type) will get thrown.